### ISSUE TITLE
Knobs to improve imbalanced data

### ISSUE THREAD

**Issue by mangleddata** [uber/ludwig#592]
I have an imbalanced classifier data set (80% True, 20% False) which is yielding interesting results. To start off, I oversample the train csv, but not valid/test csv. For whatever reason, there is a significant variance between 1st and 2nd epoch with the valid/test dataset (consistently drops like below, which seems strange since every run is fresh and I would expect some randomness). It coverges too soon (1st epoch) because the combined loss doesn't make much progress. 

<img width="652" alt="Screen Shot 2019-12-04 at 5 53 01 PM" src="https://user-images.githubusercontent.com/57873718/70197200-e7b22d80-16bf-11ea-85ef-617eac90968c.png">

The true_positive_rate for "True" class is ~0.17 and true_positive_rate for "False" class is ~0.81 ["False" class f1 score = ~0.74, "True" class f1 score = ~.021 ] - so it isn't learning "True" classifier.

Any tips on how I could improve ? Oversample valid set (which feels like cheating!), run longer epochs is something I am looking into. But are there other things I should consider ?

Also thank you for open sourcing this. It's been so much fun for wannabe practitioners like me! I also understand the question isn't so much tied to ludwig per se, but would greatly appreciate specific knobs that others have tried and tuned. The dataset is mostly binary/category columns with couple of numerical and one text column FWIW.

**Comment by mangleddata** [comment]
I made progress by reducing the default learning rate 100x (set to 0.00001). Thought I would post it as it may assist others. Closing it since progress has been made.

**Comment by w4nderlust** [comment]
The scores of the loss seems really high at the seocnd epoch, so it's likely a convergence problem. Decreasing the learning rate is usually a good starting point, glad it worked for you, but I'm curious if there was something specific in your data that led to this weird behavior. Anyway, there are some knobs for unbalanced data, like label smoothing and class weighting that can turn out to be useful for you.

**Comment by mangleddata** [comment]
There were some bad outliers in my data which I removed (bad as in unrealistic outliers). I don't know if that's what caused this or not - but since then my data has been cleaned. But still hunting on ideas to improve False Discovery Rate. What I need is to reduce the false discovery rate of one of the class  (but could compromise on other metrics for example). I also oversampled the training data (for example 4x times so the # of datapoints in both the classes are same in train set - but valid/test is kept same). This helped a bit. 

Thanks for the tip on label smoothing. Looking into it now/
